custom_components/plugwise/models.py

```diff
@@ -407,46 +391,43 @@ class PlugwiseBinarySensorEntityDescription(
         icon="mdi:valve",
         entity_category=EntityCategory.DIAGNOSTIC,
         native_unit_of_measurement=PERCENTAGE,
-        value_fn=lambda data: data["sensors"]["valve_position"],
     ),
     PlugwiseSensorEntityDescription(
         key="water_pressure",
         translation_key="water_pressure",
         device_class=SensorDeviceClass.PRESSURE,
         entity_category=EntityCategory.DIAGNOSTIC,
         native_unit_of_measurement=UnitOfPressure.BAR,
-        value_fn=lambda data: data["sensors"]["water_pressure"],
     ),
     PlugwiseSensorEntityDescription(
         key="relative_humidity",
         translation_key="relative_humidity",
         device_class=SensorDeviceClass.HUMIDITY,
         native_unit_of_measurement=PERCENTAGE,
-        value_fn=lambda data: data["sensors"]["humidity"],
+        pw_key="humidity",
     ),
     PlugwiseSensorEntityDescription(
         key="dhw_temperature",
         translation_key="dhw_temperature",
         device_class=SensorDeviceClass.TEMPERATURE,
         entity_category=EntityCategory.DIAGNOSTIC,
         native_unit_of_measurement=UnitOfTemperature.CELSIUS,
-        value_fn=lambda data: data["sensors"]["dhw_temperature"],
     ),
     PlugwiseSensorEntityDescription(
         key="domestic_hot_water_setpoint",
         translation_key="domestic_hot_water_setpoint",
         device_class=SensorDeviceClass.TEMPERATURE,
         entity_category=EntityCategory.DIAGNOSTIC,
         native_unit_of_measurement=UnitOfTemperature.CELSIUS,
-        value_fn=lambda data: data["sensors"]["domestic_hot_water_setpoint"],
     ),
     PlugwiseSensorEntityDescription(
         key="maximum_boiler_temperature",
         translation_key="maximum_boiler_temperature",
         device_class=SensorDeviceClass.TEMPERATURE,
         entity_category=EntityCategory.DIAGNOSTIC,
         native_unit_of_measurement=UnitOfTemperature.CELSIUS,
-        value_fn=lambda data: data["maximum_boiler_temperature"]["setpoint"],  # type: ignore [index]
+        pw_key="setpoint",
+        pw_lookup="maximum_boiler_temperature",
     ),
```

Comment:
I think we should remove this one.
This one will always be a select, never a sensor.

Reply:
The rest looks very good!

Reply:
Should this construct be used for `number` and `select` as well?

Reply:
If we don't need the lookup for 'other than sensors' dict we can remove part of the lookup function. I was tempted to do select as well but stopped in time to first make sure it was Mypy-compatible (which took another hour before able to commit)